src/librustc_trans/mir/mod.rs

```diff
@@ -108,6 +108,16 @@ enum TempRef<'tcx> {
     Operand(Option<OperandRef<'tcx>>),
 }
 
+impl<'tcx> TempRef<'tcx> {
+    fn new_operand(val: OperandValue, ty: ty::Ty<'tcx>) -> TempRef<'tcx> {
```

Comment:
I was thinking that it would check for `ty` being zero-sized and returned either the wrapped `C_nil` or `None`, like the whole `else-if-else` block did in the original commit.